security/keycloak-oidc-client-basic/src/main/java/io/quarkus/ts/security/keycloak/oidcclient/basic/TokenProviderResource.java

```diff
@@ -40,6 +44,12 @@ public String getTokenUsingAdminUserPasswordGrant() {
         return generateToken(allOidcClients.getClient("admin-user"));
     }
 
+    @GET
+    @Path("/token-provider")
+    public String getTokenUsingTokenProvider() {
+        return tokenProvider.getAccessToken().await().indefinitely();
```

Comment:
(Just curiosity) Why did you choose here this` .await().indefinitely();` ?

Reply:
`getAccessToken()` returning `Uni<String>` so it's used to transform it to string

Similar thing is donefor getting token from OidcClient it self

```java
private String generateToken(OidcClient client) {
    return client.getTokens().await().indefinitely().getAccessToken();
}
```